needy/library.py

```diff
@@ -106,50 +106,58 @@ def build(self):
 
         self.__log_env_overrides(env_overrides)
         with OverrideEnvironment(env_overrides):
-            post_clean_commands = configuration['post-clean'] if 'post-clean' in configuration else []
-            with cd(self.project_root()):
-                for cmd in self.evaluate(post_clean_commands):
-                    command(cmd)
+            self.__clean()
 
-            definition = ProjectDefinition(self.target(), self.project_root(), configuration)
-            project = self.project(definition)
+            project = self.project(ProjectDefinition(self.target(), self.project_root(), configuration))
+            if not project:
+                raise RuntimeError('unknown project type')
 
             unrecognized_configuration_keys = set(configuration.keys()) - project.configuration_keys() - self.additional_project_configuration_keys()
             if len(unrecognized_configuration_keys):
                 raise RuntimeError('unrecognized project configuration keys: {}'.format(', '.join(unrecognized_configuration_keys)))
 
             project.set_string_format_variables(**self.string_format_variables())
 
-            if not project:
-                raise RuntimeError('unknown project type')
```

Comment:
This check appears to come a little bit too late--after `project.configuration_keys` and `project. set_string_format_variables` are called. Bumped it up a little.